From cc9f8102859cccd95ba7a13c00bc3cc4fb53aa0a Mon Sep 17 00:00:00 2001
From: Rick Harris <rickfharris@yahoo.com.au>
Date: Sun, 15 Apr 2012 09:44:55 +0000
Subject: [PATCH] dhcp: do not require dhclient-4 to be built with DHCPv6

The "-4" flag is the default if dhclient-4 was built with DHCPv6 support
enabled, and it's invalid if dhclient-4 was built without DHCPv6.
Therefore, we simply should not use it.

https://bugzilla.gnome.org/show_bug.cgi?id=679428
---
 src/dhcp-manager/nm-dhcp-dhclient.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/dhcp-manager/nm-dhcp-dhclient.c b/src/dhcp-manager/nm-dhcp-dhclient.c
index 043cd63..14be5b2 100644
--- a/src/dhcp-manager/nm-dhcp-dhclient.c
+++ b/src/dhcp-manager/nm-dhcp-dhclient.c
@@ -490,9 +490,10 @@ dhclient_start (NMDHCPClient *client,
 		g_ptr_array_add (argv, (gpointer) "-6");
 		if (mode_opt)
 			g_ptr_array_add (argv, (gpointer) mode_opt);
-	} else {
-		g_ptr_array_add (argv, (gpointer) "-4");
 	}
+        /* We never want the "-4" flag: it's the default if dhclient was
+         * built with DHCPv6 support, and invalid otherwise.
+         */
 #endif
 
 	g_ptr_array_add (argv, (gpointer) "-sf");	/* Set script file */
-- 
1.7.8.6

